pkg/lvm/lvm_util.go

```diff
@@ -1356,3 +1457,101 @@ func removeVolumeFilesystem(lvmVolume *apis.LVMVolume) error {
+	// Skip UUID change for ext4/ext3 or block devices without filesystem
+	if fsType == "ext4" || fsType == "ext3" || fsType == "" {
+		klog.Infof("Skipping UUID update for %s (fs: %s)", devicePath, fsType)
+		return nil
+	}
+
+	// update volume UUID based on filesystem type
+	switch fsType {
+	case "btrfs":
+		err := updateBtrfsUUID(devicePath)
+		if err != nil {
+			klog.Errorf("failed to update device btrfs filesystem %s UUID, error: %v", devicePath, err)
+			return err
+		}
+		klog.Infof("Successfully updated btrfs filesystem UUID for device %s", devicePath)
+	case "xfs":
+		err := updateXfsUUID(devicePath)
+		if err != nil {
+			klog.Errorf("failed to update device xfs filesystem %s UUID, error: %v", devicePath, err)
+			return err
+		}
+		klog.Infof("Successfully updated xfs filesystem UUID for device %s", devicePath)
+	default:
+		klog.Errorf("unsupported filesystem type %s for device %s", fsType, devicePath)
+		return fmt.Errorf("unsupported filesystem type %s for device %s", fsType, devicePath)
+	}
+	return nil
+}
+
+// detectFsType will detect filesystem type for given device path
+func detectFsType(devicePath string) (string, error) {
+	// get only TYPE of the filesystem
+	// Command: blkid -s TYPE -o value /dev/lvmvg/pvc-14e7c929-db3f-4335-b492-485ac6e56454
+	out, _, err := RunCommandSplit(BlockIdCommand, "-s", "TYPE", "-o", "value", devicePath)
+	if err != nil {
+		klog.Errorf("failed to get device %s TYPE, error: %v ,out: %s", devicePath, err, string(out))
+		return "", err
+	}
+	klog.V(4).Infof("Successfully fetched device %s TYPE %s", devicePath, strings.TrimSpace(string(out)))
+	return strings.TrimSpace(string(out)), nil
+}
+
+// updateBtrfsUUID update UUID of btrfs filesystem for given device path
+// command: yes | btrfstune -u /dev/lvmvg/pvc-d16fa9be-806d-4eb2-ae80-fd35e67a3483
+func updateBtrfsUUID(devicePath string) error {
+	cmd := exec.Command(BtrfstuneCommand, "-u", devicePath)
+	// Get stdin pipe to send response
+	stdin, err := cmd.StdinPipe()
+	if err != nil {
+		return fmt.Errorf("failed to get stdin pipe: %w", err)
+	}
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	cmd.Stderr = &out
+	// Start the command
+	if err := cmd.Start(); err != nil {
+		return fmt.Errorf("failed to start btrfstune: %w", err)
+	}
+	// Send EXACTLY "y" to stdin to confirm UUID change
+	_, err = stdin.Write([]byte("y"))
+	if err != nil {
+		_ = cmd.Process.Kill()
+		return fmt.Errorf("failed to send 'y': %v", err)
+	}
+	stdin.Close() // Critical: unblocks the process
+	// Wait for completion
+	if err := cmd.Wait(); err != nil {
+		return fmt.Errorf("btrfstune failed: %v\nOutput: %s", err, out.String())
+	}
+	klog.V(4).Infof("Btrfs UUID updated: %s", out.String())
+	return nil
+}
+
+// updateXfsUUID updates the UUID of an XFS filesystem using xfs_admin -U generate
+// command: xfs_admin -U generate: generates and applies a new UUID directly
+func updateXfsUUID(devicePath string) error {
```

Comment:
We already have code for these. Ex [xfs](https://github.com/openebs/lib-csi/blob/e21bd9259fa6e97350b2326dc932dd2958a90e7e/pkg/xfs/xfs_util.go#L96)

Note that the code is different, ex `xfs` is replaying the log which is probably needed